src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.dev.cinema.service;
+
+import com.dev.cinema.exception.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent()) {
+            String hashedPassword = HashUtil.hashPassword(password, user.get().getSalt());
+            if (hashedPassword.equals(user.get().getPassword())) {
+                return user.get();
+            }
+        }
+        throw new AuthenticationException("Can't authenticate by email " + email
```

Comment:
don't show back the credentials. you might reveal the password while input field hides it